app/src/main/java/com/example/sagar/popupshops_buyerside/MainActivity.java

```diff
@@ -126,7 +172,12 @@ public void onKeyEntered(String key, GeoLocation location) {
                     public void onChildAdded(DataSnapshot dataSnapshot, String s) {
                         Log.d(TAG, "onChildAdded:" + dataSnapshot.getKey());
                         Item item = dataSnapshot.getValue(Item.class);
-                        mSwipeView.addView(new ItemCard(mContext, item, mSwipeView));
+                        final Spinner dropdown = (Spinner) findViewById(R.id.spinner1);
+                        items.add(item);
+                        String selectedItem = dropdown.getSelectedItem().toString();
+                        if(selectedItem == "All" || (item.getItemCategory() != null && item.getItemCategory().equals(selectedItem))){
+                            mSwipeView.addView(new ItemCard(mContext, item, mSwipeView));
+                        }
                     }
```

Comment:
This file is to do with the main buyer activity. But the layout rules are weird so cant select dropdown